services/cd-service/pkg/repository/repository_test.go

```diff
@@ -693,87 +693,92 @@ func TestConfigValidity(t *testing.T) {
 
 func TestGc(t *testing.T) {
 	tcs := []struct {
-		Name          string
-		GcFrequencies []uint
-		CreateGarbage func(t *testing.T, repo *repository)
-		Test          func(t *testing.T, repos []*repository)
+		Name               string
+		GcFrequency        uint
+		DisableSqlite      bool
+		ExpectedGarbageMin uint64
+		ExpectedGarbageMax uint64
 	}{
 		{
-			Name: "simple",
-			GcFrequencies: []uint{
-				// 0 disables GC entirely
-				0,
-				// we are going to perform 101 requests, that should trigger a gc
-				101,
-			},
-			CreateGarbage: func(t *testing.T, repo *repository) {
-				ctx := context.Background()
-				err := repo.Apply(ctx, &CreateEnvironment{
-					Environment: "test",
-				})
-				if err != nil {
-					t.Fatal(err)
-				}
-				for i := 0; i < 100; i++ {
-					err := repo.Apply(ctx, &CreateApplicationVersion{
-						Application: "test",
-						Manifests: map[string]string{
-							"test": fmt.Sprintf("test%d", i),
-						},
-					})
-					if err != nil {
-						t.Fatal(err)
-					}
-				}
-			},
-			Test: func(t *testing.T, repos []*repository) {
-				ctx := context.Background()
-				stats0, err := repos[0].countObjects(ctx)
-				if err != nil {
-					t.Fatal(err)
-				}
-				if stats0.Count == 0 {
-					t.Errorf("expected object count to not be 0, but got %d", stats0.Count)
-				}
-				stats1, err := repos[1].countObjects(ctx)
-				if err != nil {
-					t.Fatal(err)
-				}
-				if stats1.Count != 0 {
-					t.Errorf("expected object count to be 0, but got %d", stats1.Count)
-				}
-			},
+			// 0 disables GC entirely
+			// we are reasonably expecting some additional files around
+			Name:               "gc disabled",
+			GcFrequency:        0,
+			DisableSqlite:      true,
+			ExpectedGarbageMin: 907,
+			ExpectedGarbageMax: 910,
+		},
+		{
+			// we are going to perform 101 requests, that should trigger a gc
+			// the number of additional files should be lower than in the case above
+			Name:               "gc enabled",
```

Comment:
How are 101 requests being done here? And how does the expected garbage stays limited between 9 and 10 exactly? I am not knowledgable in sqlite

Reply:
The 101 requests are performed later in the test block. In that testcase sqlite is even disabled and test case was already present before my change. The garbage is collected in this case by having git itself doing compactions. This is exactly the process that fails sometimes.

Reply:
Better yet, how do we tell git to compact the garbage files?

Reply:
That's happening here: https://github.com/freiheit-com/kuberpult/blob/main/services/cd-service/pkg/repository/repository.go#L676